tests/link/pytorch/test_shape.py

```diff
@@ -0,0 +1,72 @@
+import numpy as np
+
+import pytensor.tensor as pt
+from pytensor.compile.ops import DeepCopyOp, ViewOp
+from pytensor.configdefaults import config
+from pytensor.graph.fg import FunctionGraph
+from pytensor.tensor.shape import Shape, Shape_i, Unbroadcast, reshape
+from pytensor.tensor.type import iscalar, vector
+from tests.link.pytorch.test_basic import compare_pytorch_and_py
+
+
+def test_pytorch_shape_ops():
+    x_np = np.zeros((20, 3))
+    x = Shape()(pt.as_tensor_variable(x_np))
+    x_fg = FunctionGraph([], [x])
+
+    compare_pytorch_and_py(x_fg, [], must_be_device_array=False)
+
+    x = Shape_i(1)(pt.as_tensor_variable(x_np))
+    x_fg = FunctionGraph([], [x])
+
+    compare_pytorch_and_py(x_fg, [], must_be_device_array=False)
+
+
+def test_pytorch_specify_shape():
+    in_pt = pt.matrix("in")
+    x = pt.specify_shape(in_pt, (4, None))
+    x_fg = FunctionGraph([in_pt], [x])
+    compare_pytorch_and_py(x_fg, [np.ones((4, 5)).astype(config.floatX)])
+
+    # When used to assert two arrays have similar shapes
+    in_pt = pt.matrix("in")
+    shape_pt = pt.matrix("shape")
+    x = pt.specify_shape(in_pt, shape_pt.shape)
+    x_fg = FunctionGraph([in_pt, shape_pt], [x])
+    compare_pytorch_and_py(
+        x_fg,
+        [np.ones((4, 5)).astype(config.floatX), np.ones((4, 5)).astype(config.floatX)],
+    )
+
+
+def test_pytorch_Reshape_constant():
+    a = vector("a")
+    x = reshape(a, (2, 2))
+    x_fg = FunctionGraph([a], [x])
+    compare_pytorch_and_py(x_fg, [np.r_[1.0, 2.0, 3.0, 4.0].astype(config.floatX)])
+
+
+def test_pytorch_Reshape_shape_graph_input():
+    a = vector("a")
+    shape_pt = iscalar("b")
+    x = reshape(a, (shape_pt, shape_pt))
+    x_fg = FunctionGraph([a, shape_pt], [x])
+    compare_pytorch_and_py(x_fg, [np.r_[1.0, 2.0, 3.0, 4.0].astype(config.floatX), 2])
+
+
+def test_pytorch_compile_ops():
+    x = DeepCopyOp()(pt.as_tensor_variable(1.1))
+    x_fg = FunctionGraph([], [x])
```

Comment:
We have to make sure DeepCopy does the expected thing. For instance here is how we know it is currently not doing the right thing in the Numba backend: https://github.com/pymc-devs/pytensor/issues/50

Reply:
Is the `DeepCopyOp` actually in the scope of these changes?
Not sure why `DeepCopy` and `View` were being tested together with `Unbroadcast` (I adapted the tests from the JAX backend implementation).

Reply:
You can leave then out if you didn't mean to implement it

Reply:
Fwiw, i tested the torch clone and it _seems_ like it's fine

```
import pytensor
from pytensor import tensor as pt

x = pytensor.shared(0, name="x")
f = pytensor.function([], x, mode=None)
f().itemset(2)
assert x.get_value() == 0

f = pytensor.function([], x, mode="PYTORCH")
f().apply_(lambda _: 2)
assert x.get_value() == 0

```